activemodel/test/cases/errors_test.rb

```diff
@@ -86,47 +85,55 @@ def test_no_key
   end
 
   test "error access is indifferent" do
-    errors = ActiveModel::Errors.new(self)
-    errors[:foo] << "omg"
+    errors = ActiveModel::Errors.new(Person.new)
+    errors.add(:name, "omg")
 
-    assert_equal ["omg"], errors["foo"]
+    assert_equal ["omg"], assert_deprecated { errors["name"] }
   end
 
   test "values returns an array of messages" do
     errors = ActiveModel::Errors.new(self)
-    errors.messages[:foo] = "omg"
-    errors.messages[:baz] = "zomg"
+    assert_deprecated { errors.messages[:foo] = "omg" }
```

Comment:
@matthewd About the to_hash's `[] =` case, it was probably not mentioned in the doc, but this spec provides direct usage. So unless we remove this spec, we had to tempter with `to_hash`.